core/src/main/java/com/graphhopper/routing/util/spatialrules/countries/GermanySpatialRule.java

```diff
@@ -17,46 +17,81 @@
  */
 package com.graphhopper.routing.util.spatialrules.countries;
 
+import java.util.List;
+
+import org.locationtech.jts.geom.Polygon;
+
 import com.graphhopper.routing.profiles.Country;
 import com.graphhopper.routing.profiles.RoadAccess;
-import com.graphhopper.routing.util.spatialrules.DefaultSpatialRule;
+import com.graphhopper.routing.profiles.RoadClass;
+import com.graphhopper.routing.util.spatialrules.AbstractSpatialRule;
 import com.graphhopper.routing.util.spatialrules.TransportationMode;
 
 /**
  * Defines the default rules for German roads
  *
  * @author Robin Boldt
  */
-public class GermanySpatialRule extends DefaultSpatialRule {
+public class GermanySpatialRule extends AbstractSpatialRule {
+    
+    public GermanySpatialRule(List<Polygon> borders) {
+        super(borders);
+    }
 
     /**
      * Germany contains roads with no speed limit. For these roads, this method will return Integer.MAX_VALUE.
      * Your implementation should be able to handle these cases.
      */
     @Override
-    public double getMaxSpeed(String highwayTag, double _default) {
+    public double getMaxSpeed(RoadClass roadClass, TransportationMode transport, double currentMaxSpeed) {
+        if (currentMaxSpeed > 0 || transport != TransportationMode.MOTOR_VEHICLE) {
+            return currentMaxSpeed;
+        }
+        
         // As defined in: https://wiki.openstreetmap.org/wiki/OSM_tags_for_routing/Maxspeed#Motorcar
-        switch (highwayTag) {
-            case "motorway":
-            case "trunk":
+        switch (roadClass) {
+            case MOTORWAY:
+            case TRUNK:
                 return Integer.MAX_VALUE;
```

Comment:
Should we make it compatible with our MaxSpeed EncodedValue and return MaxSpeed.UNLIMITED_SIGN_SPEED?